pkg/ipamutil/reconciler.go

```diff
@@ -245,6 +247,26 @@ func (r *ClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ct
 		return unwrapResult(res), err
 	}
 
+	// We need to ensure the address is properly watched by controller-runtime client (as it is cached) before moving
+	// to the next reconciliation request.
+	// This is required, otherwise if the next "EnsureAddress" is called too fast, or the controller-runtime cached client informer
+	// is too slow, we may end up not seeing that the IP Address is being used and can provide duplicated IPs.
+	err = wait.PollUntilContextTimeout(ctx, 5*time.Millisecond, 5*time.Second, true, func(ctx context.Context) (bool, error) {
+		key := client.ObjectKeyFromObject(&address)
+		if err := r.Client.Get(ctx, key, &ipamv1.IPAddress{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
```

Comment:
nit: we could use `return false, client.IgnoreNotFound(err)` here instead

Reply:
fixed and rebased already :) 

lmk if this is the request! thanks!